From 3a2ac6b9647c254eda535a3a23cb495128b32099 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:53:03 +0800
Subject: [PATCH] Revert "mmc: Add new API call 'sdio_reset_comm' for resetting
 communication with an SDIO device"

This reverts commit b6c5dbe49bdfd8f814243e7b7922167e9fbc853b.
---
 drivers/mmc/core/sdio.c | 57 -----------------------------------------
 1 file changed, 57 deletions(-)

diff --git a/drivers/mmc/core/sdio.c b/drivers/mmc/core/sdio.c
index ebfdea444f42..f823bf793347 100644
--- a/drivers/mmc/core/sdio.c
+++ b/drivers/mmc/core/sdio.c
@@ -616,60 +616,3 @@ int mmc_attach_sdio(struct mmc_host *host, u32 ocr)
 	return err;
 }
 
-int sdio_reset_comm(struct mmc_card *card)
-{
-	struct mmc_host *host = card->host;
-	u32 ocr;
-	int err;
-
-	printk("%s():\n", __func__);
-	mmc_go_idle(host);
-
-	mmc_set_clock(host, host->f_min);
-
-	err = mmc_send_io_op_cond(host, 0, &ocr);
-	if (err)
-		goto err;
-
-	host->ocr = mmc_select_voltage(host, ocr);
-	if (!host->ocr) {
-		err = -EINVAL;
-		goto err;
-	}
-
-	err = mmc_send_io_op_cond(host, host->ocr, &ocr);
-	if (err)
-		goto err;
-
-	if (mmc_host_is_spi(host)) {
-		err = mmc_spi_set_crc(host, use_spi_crc);
-		if (err)
-		goto err;
-	}
-
-	if (!mmc_host_is_spi(host)) {
-		err = mmc_send_relative_addr(host, &card->rca);
-		if (err)
-			goto err;
-		mmc_set_bus_mode(host, MMC_BUSMODE_PUSHPULL);
-	}
-	if (!mmc_host_is_spi(host)) {
-		err = mmc_select_card(card);
-		if (err)
-			goto err;
-	}
-
-	mmc_set_clock(host, card->cis.max_dtr);
-	err = sdio_enable_wide(card);
-	if (err)
-		goto err;
-
-	return 0;
- err:
-	printk("%s: Error resetting SDIO communications (%d)\n",
-	       mmc_hostname(host), err);
-	return err;
-}
-EXPORT_SYMBOL(sdio_reset_comm);
-
-
-- 
2.35.3

